components/butterfly/src/main.rs

```diff
@@ -32,19 +32,37 @@ fn main() {
     let bind_to = args.next().unwrap();
     println!("Binding to {}", bind_to);
 
-    let server = server::Server::new(&bind_to[..], member::Member::new(), trace::Trace::default())
-        .unwrap();
-    println!("Server ID: {}", server.member.read().unwrap().get_id());
+    let bind_to_addr = bind_to.parse::<SocketAddr>().unwrap();
+    let bind_port = bind_to_addr.port();
+    let mut gossip_bind_addr = bind_to_addr.clone();
+    let gport = bind_port + 1;
+    gossip_bind_addr.set_port(gport);
+
+    let mut member = member::Member::new();
+    member.set_swim_port(bind_port as i32);
+    member.set_gossip_port(gport as i32);
 
-    server.start(server::outbound::Timing::default()).expect("Cannot start server");
+    let server = server::Server::new(bind_to_addr,
+                                     gossip_bind_addr,
+                                     member,
+                                     trace::Trace::default(),
+                                     None)
+        .unwrap();
+    println!("Server ID: {}", server.member_id);
 
     let targets: Vec<String> = args.collect();
     for target in &targets {
         let addr: SocketAddr = target.parse().unwrap();
-        let member = member::Member::from(addr);
-        server::outbound::ping(&server, &member, addr, None);
+        let mut member = member::Member::new();
+        member.set_address(format!("{}", addr.ip()));
+        member.set_swim_port(addr.port() as i32);
+        member.set_gossip_port(addr.port() as i32);
```

Comment:
Good eye! No, it's not. It doesn't matter in this case (which is why it worked at all) because all the `add_initial_members` function really does is fire a single SWIM packet into the ether. Fix inbound.